docs/contributor/devenv_setup.md

```diff
@@ -307,3 +307,137 @@ sudo lvextend -l +95%FREE /dev/mapper/rhel-root
+the directory from the virtual machine.
+
+```
+BRIDGE_DEV=$(sudo virsh net-info default | grep '^Bridge:' | awk '{print $2}')
+BRIDGE_IP=$(ip -f inet addr show "${BRIDGE_DEV}" | grep inet)
+
+echo "${BRIDGE_IP}" | awk '{print $2}' | cut -d/ -f1
+```
+
+#### Virtual Machine Configuration
+
+```
+MOUNT_DIR=/home/microshift-shared
+BRIDGE_IP=192.168.100.1
+
+sudo dnf install -y nfs-utils
+
+sudo mkdir -p "${MOUNT_DIR}"
+sudo mount -t nfs4 -o context="system_u:object_r:container_file_t:s0" "${BRIDGE_IP}:${MOUNT_DIR}" "${MOUNT_DIR}"
+```
+
+If the mount works properly, make it permanent by running the following command.
+
+```
+echo "${BRIDGE_IP}:${MOUNT_DIR} ${MOUNT_DIR} nfs4 rw,hard,intr,noatime,context=\"system_u:object_r:container_file_t:s0\" 0 0" | sudo tee -a /etc/fstab
+```
+
+#### Synchronize User and Group Identifiers
+
+For read-write to work properly both on the hypervisor host and on the virtual
+machine, it is recommended to edit the `microshift` user and group
+identifiers on the virtual machine to have the value as on the hypervisor.
+
+Edit the `/etc/passwd` and `/etc/group` files on the virtual machine to replace
+the `microshift` user and group identifiers. The goal is for the `id -u; id -g`
+command to return the same identifiers when run on the host and on the virtual
+machine.
+
+Here is an example of setting the user and group identifiers to `12345` value on
+the virtual machine.
+
+```
+$ grep ^microshift /etc/passwd
+microshift:x:12345:12345::/home/microshift:/bin/bash
+
+$ grep ^microshift /etc/group
+microshift:x:12345:
+```
+
+Run the following commands to fix the ownership of the user home directory and
+the shared directory.
+
+```
+MOUNT_DIR=/home/microshift-shared
+
+sudo chown -R $(whoami). ~
+sudo chown -R $(whoami). "${MOUNT_DIR}"
+
+# Delete user-specific container cache
+sudo rm -rf ~/.local/share/containers/
+```
+
+Reboot the development virtual machine to finalize the configuration.
+
+#### Using Shared Directory
+
+One way of using the shared directory would be to keep the whole source tree in
+it for easy access from the host.
+
+```
+MOUNT_DIR=/home/microshift-shared
+
+sudo mkdir -p -m 0755 "${MOUNT_DIR}/microshift"
+sudo chown $(whoami). "${MOUNT_DIR}/microshift"
+
+# Do not attempt to create the link if the target already exists
+[ ! -e ~/microshift ] && ln -s "${MOUNT_DIR}/microshift" ~/microshift
+```
+
+Another way of using the shared directory would be to keep the whole `microshift`
+user home directory on the host and mount it from the virtual machine.
+
+```
+MOUNT_DIR=/home/microshift-shared
+
+sudo mkdir -p -m 0755 "${MOUNT_DIR}/microshift"
+sudo chown $(whoami). "${MOUNT_DIR}/microshift"
+
+# Move the existing home directory
+[ -d /home/microshift ] && sudo mv /home/microshift /home/microshift.local
+# Do not attempt to create the link if the target already exists
+[ ! -L /home/microshift ] && sudo ln -s "${MOUNT_DIR}" /home/microshift
```

Comment:
I think this goes too far. I don't think it's a good advice to move existing home directory and mount whole share as home dir. What about all the dotfiles there, like ~/.config, ~/.ssh, ~/.local/bin, and others?